src/private.h

```diff
@@ -264,6 +264,53 @@ GPUARRAY_LOCAL void gpukernel_source_with_line_numbers(unsigned int count,
                                                        size_t *newl,
                                                        strb *src);
 
+static uint16_t float_to_half(float value) {
```

Comment:
why this is needed? I think using the native operation from cuda/opencl is better then using this.